cni/network/invoker_cns.go

```diff
@@ -132,97 +133,44 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
 			hostPrimaryIP:      response.PodIPInfo[i].HostPrimaryIPInfo.PrimaryIP,
 			hostGateway:        response.PodIPInfo[i].HostPrimaryIPInfo.Gateway,
-		}
-
-		// set the NC Primary IP in options
-		// SNATIPKey is not set for ipv6
-		if net.ParseIP(info.ncPrimaryIP).To4() != nil {
-			addConfig.options[network.SNATIPKey] = info.ncPrimaryIP
+			nicType:            response.PodIPInfo[i].NICType,
+			macAddress:         response.PodIPInfo[i].MacAddress,
+			skipDefaultRoutes:  response.PodIPInfo[i].SkipDefaultRoutes,
+			routes:             response.PodIPInfo[i].Routes,
 		}
 
 		logger.Info("Received info for pod",
-			zap.Any("ipv4info", info),
+			zap.Any("ipinfo", info),
 			zap.Any("podInfo", podInfo))
-		ip, ncIPNet, err := net.ParseCIDR(info.podIPAddress + "/" + fmt.Sprint(info.ncSubnetPrefix))
-		if ip == nil {
-			return IPAMAddResult{}, errors.Wrap(err, "Unable to parse IP from response: "+info.podIPAddress+" with err %w")
-		}
 
-		ncgw := net.ParseIP(info.ncGatewayIPAddress)
-		if ncgw == nil {
-			// TODO: Remove v4overlay and dualstackoverlay options, after 'overlay' rolls out in AKS-RP
-			if (invoker.ipamMode != util.V4Overlay) && (invoker.ipamMode != util.DualStackOverlay) && (invoker.ipamMode != util.Overlay) {
-				return IPAMAddResult{}, errors.Wrap(errInvalidArgs, "%w: Gateway address "+info.ncGatewayIPAddress+" from response is invalid")
-			}
-
-			if net.ParseIP(info.podIPAddress).To4() != nil { //nolint:gocritic
-				ncgw, err = getOverlayGateway(ncIPNet)
-				if err != nil {
-					return IPAMAddResult{}, err
-				}
-			} else if net.ParseIP(info.podIPAddress).To16() != nil {
-				ncgw = net.ParseIP(overlayGatewayV6IP)
-			} else {
-				return IPAMAddResult{}, errors.Wrap(err, "No podIPAddress is found: %w")
+		//nolint:exhaustive // ignore exhaustive types check
+		switch info.nicType {
+		case cns.DelegatedVMNIC:
+			if !info.skipDefaultRoutes {
+				numInterfacesWithDefaultRoutes++
 			}
-		}
-
-		// construct ipnet for result
-		resultIPnet := net.IPNet{
-			IP:   ip,
-			Mask: ncIPNet.Mask,
-		}
 
-		if net.ParseIP(info.podIPAddress).To4() != nil {
-			addResult.ipv4Result = &cniTypesCurr.Result{
-				IPs: []*cniTypesCurr.IPConfig{
-					{
-						Address: resultIPnet,
-						Gateway: ncgw,
-					},
-				},
-				Routes: []*cniTypes.Route{
-					{
-						Dst: network.Ipv4DefaultRouteDstPrefix,
-						GW:  ncgw,
-					},
-				},
+			if err := configureSecondaryAddResult(&info, &addResult, &response.PodIPInfo[i].PodIPConfig); err != nil {
+				return IPAMAddResult{}, err
 			}
-		} else if net.ParseIP(info.podIPAddress).To16() != nil {
-			addResult.ipv6Result = &cniTypesCurr.Result{
-				IPs: []*cniTypesCurr.IPConfig{
-					{
-						Address: resultIPnet,
-						Gateway: ncgw,
-					},
-				},
-				Routes: []*cniTypes.Route{
-					{
-						Dst: network.Ipv6DefaultRouteDstPrefix,
-						GW:  ncgw,
-					},
-				},
+		default:
+			// only count dualstack interface once
```

Comment:
Is it typo? default interface or dualstack interface?

Reply:
since we get v4 and v6 pod ip info separately for dualstack, we just want to count this interface once

Reply:
ok then why are we not doing similar check for secondary interface?

Reply:
we're not handling v6 for secondary interface at the moment

not sure if we want to change the current behaviour where the v4 and v6 ip info are sent separately for the same interface